diffkemp/simpll/ModuleComparator.cpp

```diff
@@ -62,10 +76,16 @@ void ModuleComparator::compareFunctions(Function *FirstFun,
                                          showAsmDiffs, &GS, DI, this);
     if (fComp.compare() == 0) {
         DEBUG_WITH_TYPE(DEBUG_SIMPLL,
-                        dbgs() << "Function " << FirstFun->getName()
-                               << " is same in both modules\n");
+            decreaseDebugIndentLevel();
```

Comment:
You are duplicating code here (calling `decreaseDebugIndentLevel` both here and in the `else` branch). I suggest moving the call before the condition (right after the invocation of the function comparator).

Reply:
Correct, but this was intentional. The problem is that the function comparator needs to be executed before the indentation decrease. So, at least as far as I understand it, this wouldn't be possible unless I also move `fComp.compare()` before the condition and save its result into a new variable, which is in my opinion worse than one extra debug line. The same goes for the other code duplication comments below.

Reply:
Indeed, I did not realize this. Let's move `fComp.compare()` out of the condition and assign it into some variable `result`. Currently, it's kind of hidden there (I just overlooked it when writing the comment) and it decreases readability.